bpf/bpf_lxc.c

```diff
@@ -878,6 +878,7 @@ static __always_inline int handle_ipv4_from_lxc(struct __ctx_buff *ctx, __u32 *d
 
 #ifdef ENABLE_NODEPORT
 # ifdef ENABLE_DSR
+		/* only for old connections: */
```

Comment:
After upgrading Cilium v1.12 to v1.13, this code ensures that ongoing active connections will be handled appropriately. Cilium doesn't have a chance to update DSR states until clients shut down old connections and restart new connections (only the SYN packet has a clue). So this code is left here as is. Is my understanding correct?

Reply:
Ack - that's point 3 in the cover letter:

> reduce the DSR support in bpf_lxc to only cover old connections. The nodeport path can't easily create a CT entry for these connections (as for TCP we won't see another SYN with DSR info). There's possibilities how to move this into to-netdev as well, but let's try that in a follow-on PR.

We might be able to identify such old connections in the RevDNAT code @ `to-netdev`, but I'm a bit hesitant to change this for existing connections. It would be very confusing to any observer that sits "in between" `from-container` and `to-netdev` - suddenly the replies of an established flow are no longer revDNATed from their perspective.

Reply:
Agree. It's very confusing, and I think it would be a bit risky to touch the state of existing long connections.